patches/buildkit-direct-execve-v8.0/0001-linux-user-have-execve-call-qemu-via-proc-self-exe-t.patch

```diff
@@ -19,75 +19,82 @@ as argv[0] to execve.
 Signed-off-by: Tibor Vass <tibor@docker.com>
 Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>
 ---
- linux-user/syscall.c | 39 ++++++++++++++++++++++++++++++---------
- 1 file changed, 30 insertions(+), 9 deletions(-)
+ linux-user/syscall.c | 46 +++++++++++++++++++++++++++++---------------
+ 1 file changed, 31 insertions(+), 15 deletions(-)
 
 diff --git a/linux-user/syscall.c b/linux-user/syscall.c
-index 24b25759be..91d971c6fb 100644
+index 69f740ff98..d571f1c4e0 100644
 --- a/linux-user/syscall.c
 +++ b/linux-user/syscall.c
-@@ -8895,10 +8895,37 @@ static abi_long do_syscall1(CPUArchState *cpu_env, int num, abi_long arg1,
-                 envc++;
-             }
+@@ -659,8 +659,7 @@ safe_syscall4(pid_t, wait4, pid_t, pid, int *, status, int, options, \
+ #endif
+ safe_syscall5(int, waitid, idtype_t, idtype, id_t, id, siginfo_t *, infop, \
+               int, options, struct rusage *, rusage)
+-safe_syscall5(int, execveat, int, dirfd, const char *, filename,
+-              char **, argv, char **, envp, int, flags)
++safe_syscall3(int, execve, const char *, filename, char **, argv, char **, envp)
```

Comment:
Looks like this change is not needed anymore since https://github.com/qemu/qemu/commit/7b336dcd06d727c2ec98ec5dcf95f38c83ac4b9f

Reply:
Brought back in https://github.com/qemu/qemu/commit/7b336dcd06d727c2ec98ec5dcf95f38c83ac4b9f